test/files/stream.iced

```diff
@@ -0,0 +1,88 @@
+{prng} = require('crypto')
+stream = require('../../src/stream.iced')
+enc = require('../../src/encoding.iced')
+
+#==========================================================
+#Helper functions
+#==========================================================
+
+loop_limit = 5000
+loop_skip = 27
+
+stream_random_data = (strm, len) ->
+	data = prng(len)
+	i = 0
+	j = 0
+	while i < data.length
+		j = i + prng(1)[0]
+		strm.write(data[i...j])
+		i = j
+	data
+
+test_bx_consistency = (T, base, len) ->
+	encoding = encoding_for_base(base)
+	encoder = new stream.StreamEncoder(encoding)
+	decoder = new stream.StreamDecoder(encoding)
+
+	encoder.pipe(decoder)
+
+	data = stream_random_data(encoder, len)
+	decoded_data = decoder.read()
+	leftovers = encoder.read(0)
```

Comment:
so the consistency test takes a bunch of random data (returned by the stream_random_data function as a buffer for easy comparison), encodes it, then decodes it, and compares the encoded->decoded buffer against the original. The assertion would be that the two are identical, which is what line 33 handles.